app/models/miq_server/worker_management/kubernetes.rb

```diff
@@ -131,6 +144,8 @@ def ensure_kube_monitors_started
   end
 
   def delete_failed_deployments
+    return unless my_server.is_master?
+
     failed_deployments.each do |failed|
```

Comment:
failed_deployments has "global" scope (aka not scoped to a specific miq_server) so this should also be only run on the master server